content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+During this process they cannot say which of the versions is the final one.
+But since nomination process is aiming to produce a number of candidates for consensus, this is an expected state from the system consensus.
+
+In the balloting phase nodes exchange their ballots on created composites during the nomination.
+The ballots are data structures comprised of a counter and a value - the value is a candidate data and the counter is integer startign from 1.
+The balloting repeatedly makes attempts to reach a network consensus on some candiate among the group of created ones.
+This is done by repeatedly conducting federated votes on statements about ballots.
+
+The counter in each ballot tracks the attempts made and then ballots with higher counters take precedence over ballots with lower counters.
+From the point of view of a given node, consensus is reached when it finds a ballot for which it is able to confirm (find a quorum accepting) the statement `I commit to ballot X`.
+At this point it is safe to act on the value - meaning to include it in its local ledger.
+In Stellar this is also called externalizing the value.
+Once a ballot is confirmed committed, a node can be sure that every other well-behaved node has externalized, or inevitably will externalize, the same value.
+
+Using the FBA voting the Stellar network is able to achieve liveness and safety for the network, as well as to FBA consensus is the most scalable solution, first implemented by Ripple (XRP) and then improved upon by Stellar (XLM). The consensus requires a special network of nodes to be known and verified ahead of time before put into use.
+
+TODO: finish with conclusion
+
+#### Ripple Background
+
+TODO: add
+
+## Avalanche and How It Uses Sampling to Achieve Deterministic Probabilistic Consensus on a Leaderless Manner that Can Scale
+
+### Avalanche
+
+After Nakamoto consensus algorithm, the world still wanted to have an algorithm that does have its benefits like full decentralization, but also speed and energy efficiency.
+As a solutuon to this, Avalanche was proposed by a pseudonymous group called [Team Rocket](https://medium.com/@mr_davis_12/avalanche-and-the-team-rocket-chronology-mystery-and-pseudonymity-30c17331f6a2).
+[Avalanche](https://assets.website-files.com/5d80307810123f5ffbb34d6e/6009805681b416f34dcae012_Avalanche%20Consensus%20Whitepaper.pdf) is a probabilistic consensus protocol based on Proof Of Stake.
+Its most important property, which sets it apart from most of existing protocols, is that like Nakamoto consensus, it can operate with no known upper-bound of participants in the network.
+It aims to reach the decentralization level of Nakamoto, resiliance to 51% attacks, as well as high throughput and low latency.
+It follows the idea of having some neglectable error when reaching of consensus as Nakamoto suggested (P = 1 — ε), which makes it probabilistic.
```

Comment:
The idea was a simple representation of not having a 100% certainty, that is represented by 1;
and by dividing the error from it, that can be of any value, we represent that certainty size depends on the error smaller or larger.
IMO this is understandable and is explained like this in almost all theoretical papers.
Changed it here - https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R221
but not entirely sure about it

Reply:
I see what you meant by it and I think when you're familiar with it and used to consider probabilities like this, it's fine, but we shouldn't assume that from the reader IMO, even if it seems very basic, so I do prefer the new form that you introduced personally.

Maybe we can leave the thread open so that others can share their opinions, because I might be the only one to feel this way 😄 @awfm9 @Maelkum @ricardomgoncalves @emailtovamos @bogdanprodan-okta, any opinions?